src/rpm/builder.rs

```diff
@@ -804,12 +815,18 @@ impl RPMBuilder {
 
         // if we have an empty RPM, we have to leave out all file related index entries.
         if !self.files.is_empty() {
+            let (tag, data) = if small_package {
+                let file_sizes = file_sizes
+                    .into_iter()
+                    .map(u32::try_from)
+                    .collect::<Result<_, _>>()
+                    .expect("combined_file_sizes is smaller than 4 GiB");
```

Comment:
In this instance falliability is actually proper, but we shouldn't panic.  `if let Ok(file_sizes) = ... else`

Reply:
IIUC it should never panic since `combined_file_sizes` is a sum of all file sizes, thus if it's smaller than 4 GiB, it means that it's true for all files as well.

Reply:
The message is backwards here also